core/src/com/mygdx/game/models/Battle.java

```diff
@@ -0,0 +1,26 @@
+package com.mygdx.game.models;
+
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+public class Battle {
+    Set<Integer> playerIds;
+    Map<Integer, List<BasicUnit>> units;
+    private Map<Integer, Integer> strengths;
+
+    public Battle(Map<Integer, List<BasicUnit>> units) {
+
+    }
+
+    /**
+     * Resolves the battle.
+     *
+     * @return a Map of player ids to their participating units post-resolution of combat
+     */
+    public Map<Integer, List<BasicUnit>> resolveBattle() {
```

Comment:
You need to know, how much damage each unit took. And which units are retreating. May create a separate class for this which contains unit id, damage taken and wether the unit is retreating